Ios migration guide (#45)
MapsPeople/docsite: docs/01-getting-started/ios/v4/directions.mdx

---
title: Getting Directions
hide_title: false
hide_table_of_contents: true
sidebar_position: 5
---

## Getting Directions to a Location

> In a lot of cases, the user might not want to display a specific region, but rather get a route proposed that will lead them to where they need to go. In order to accomplish this, we utilize the [MPDirectionsService](https://app.mapsindoors.com/mapsindoors/reference/ios/v3/interface_m_p_directions_service.html) class, which we will be able to query through the [MPDirectionsQuery](https://app.mapsindoors.com/mapsindoors/reference/ios/v3/interface_m_p_directions_query.html) class, while to actually display the result we use the [MPDirectionsRenderer](https://app.mapsindoors.com/mapsindoors/reference/ios/v3/interface_m_p_directions_renderer.html).

Reviewer: ```suggestion
In a lot of cases, the user might not want to display a specific region, but rather get a route proposed that will lead them to where they need to go. In order to accomplish this, we utilize the [MPDirectionsService](https://app.mapsindoors.com/mapsindoors/reference/ios/v4-doc/documentation/mapsindoors/mpdirectionsservice) class, which we will be able to query using the [MPDirectionsQuery](https://app.mapsindoors.com/mapsindoors/reference/ios/v4-doc/documentation/mapsindoors/mpdirectionsquery) class, while to actually display the result we use the [MPDirectionsRenderer](https://app.mapsindoors.com/mapsindoors/reference/ios/v4-doc/documentation/mapsindoors/mpdirectionsrenderer).
```

First things first, let us add in some new variables, namely the renderer and two points we wish to acquire directional data between (the origin and destination). Usually, the origin point would refer to the location of the user, however for the purposes of demonstration we will use a static origin point.

```swift

> var origin:MPPoint = MPPoint(lat: 38.897382, lon: -77.037447, zValue:0)

Reviewer: ```suggestion
var origin= MPPoint(latitude: 38.897382, longitude: -77.037447, z: 0)
```

> var destination:MPPoint = MPPoint.init()

Reviewer: ```suggestion
var destination= MPPoint()
```

> private let renderer = MPDirectionsRenderer()

Reviewer: ```suggestion
private let renderer = mapControl.newDirectionsRenderer()
```

```

In order to update the destination variable we simply employ the same concept as we did when adding the search bar, namely we utilise the `func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath)` function. In this case, we simply need to update our function to store the point of the selected location and call `directions()`, which is a function we will add in a bit that will handle the construction of the directions query and rendering.

```swift
func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {

>         let location:MPLocation = searchResult![indexPath.row]
>         self.destination = location.getPoint()!
>         self.mapControl?.go(to: location)

Reviewer: You should try the snippets in Xcode
```suggestion
        let location = searchResult![indexPath.row]
        self.destination = location.position
        self.mapControl?.goTo(location)
```

        tableView.removeFromSuperview()
        directions()
    }
```

We can now add in the `directions()` function,

```swift
func directions() {

>     let directions = MPDirectionsService()

Reviewer: ```suggestion
    let directions = MPMapsIndoors.shared.directionsService
```
